File: javascript/3.js
const smallestPrimeFactor = n => {
  for (let x = 2; ; x++) if (!(n % x)) return x
}
const largestPrimeFactor = n => {
  const m = smallestPrimeFactor(n)
  return n === m ? n : largestPrimeFactor(n / m)
}
console.log(largestPrimeFactor(600851475143))

File: rust/3.rs
fn smallest_prime_factor(n: u64) -> u64 {
  let mut prime_factor = 2;
  while n % prime_factor != 0 {
    prime_factor += 1;
  }
  prime_factor
}

fn largest_prime_factor(n: u64) -> u64 {
  let m = smallest_prime_factor(n);
  if m == n {
    m
  } else {
    largest_prime_factor(n / m)
  }
}

fn main () {
  println!("{}", largest_prime_factor(600851475143));
}
